.github/workflows/Readme.md

```diff
@@ -0,0 +1,297 @@
+- `ACACIA_SECRET_ACCESS_KEY`: Secret access key for the S3-compatible storage.
+- `PAT_TOKEN`: PAT token is used for checking the runner label and their availability. 
+
+These variables and secrets are used throughout the workflow for authentication and tagging purposes.
+
+### Environments (`manual_approval`)
+The manual_approval environment is configured to require explicit manual approval before the deployment process proceeds. This ensures that all deployments are reviewed and authorized by designated team members, adding an additional layer of security and control to the CI pipeline.
+
+Navigate to your repository's Settings > Environments > New Environment > Name (`manual_approval`), then under Deployment protection rules, click (`Require reviewers`) and Save protection rules.
+## Workflow Overview
+
+The workflow consists of several jobs that execute sequentially or in parallel, depending on their dependencies.
+
+### PREPARE-job
+
+This initial job sets up the environment and determines whether the workflow should proceed based on the trigger logic.
+
+**Key Steps:**
+
+1. **Checkout Repository**: Uses `actions/checkout@v4` to fetch the latest code with a `fetch-depth` of 2 to ensure sufficient commit history for comparison.
+
+2. **Get Changed Files**: 
+   - Uses `git diff` to identify Dockerfiles (`*.dockerfile`) that have changed between the previous commit and the current one.
+   - Stores the list of changed files in the `files` output variable.
+
+3. **Set Proceed Flag**: 
+   - Checks if no Dockerfiles or multiple Dockerfiles have been modified.
+     - If no Dockerfiles are modified, sets `proceed_valid` to `false` and exits.
+     - If multiple Dockerfiles are modified, sets `proceed_valid` to `false` and exits.
+   - If exactly one Dockerfile is modified, proceeds to check for the required label.
+   - Verifies that the Dockerfile contains `LABEL org.opencontainers.image.compilation=auto`.
+     - If present, sets `proceed_valid` to `true`.
+     - If not, sets `proceed_valid` to `false` and exits.
+
+4. **Parse File Path**: Extracts necessary information from the modified Dockerfile:
+   - `directory`: The directory containing the Dockerfile.
+   - `dockerfile_name`: The base name of the Dockerfile without the extension.
+   - `platform` and `platform_tag`: Determined by checking the `org.opencontainers.image.arch` label in the Dockerfile.
+     - Supports `linux/arm64` (tagged as `arm`) and `linux/amd64` (tagged as `x86`).
+
+5. **Set Current Date**: Captures the current date in `MM-DD` format for use in tagging the Docker image.
+
+6. **Debug Variables**: Outputs the collected variables for verification purposes.
+
+### BUILD-AND-PUSH-job
+
+This job builds the Docker image using Docker Buildx and saves it for later use.
+
+**Key Steps:**
+
+1. **Set Up QEMU**: Uses `docker/setup-qemu-action@v3` to enable cross-platform emulation, allowing builds for different architectures.
```

Comment:
should reference that setting up the QEMU emulator and provide a link to their online documentation.

Reply:
It is updated as you indicated. The hyperlink has been given.